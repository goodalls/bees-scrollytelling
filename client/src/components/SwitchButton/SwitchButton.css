.toggle {
  margin: 10px 0;
  position: fixed;
  top: 15px;
  right: 15px;
  z-index: 10; }
  .toggle label {
    position: relative;
    display: inline-block;
    width: 80px;
    height: 40px;
    background-color: #e1e1e1;
    border-radius: 50px;
    transition: all 0.3s ease-in-out;
    cursor: pointer;
    box-shadow: inset 0 0 2px 1px rgba(0, 0, 0, 0.1);
    -webkit-tap-highlight-color: transparent; }
    .toggle label:before {
      content: "";
      position: absolute;
      top: 3px;
      left: 3px;
      height: 34px;
      width: 34px;
      background-color: #fff;
      border-radius: 46px;
      box-shadow: 1px 1px 5px rgba(0, 0, 0, 0.1);
      transition: box-shadow 0.3s ease-in-out;
      animation: moveLeft 0.3s ease-in-out; }
    .toggle label div {
      display: flex;
      justify-content: space-around;
      /*      position: absolute;
      top: -17px;
      right: 92px;
      width: fit-content;*/
      text-align: right; }
      .toggle label div p {
        margin: 10px 0; }
    .toggle label h4 {
      position: absolute;
      bottom: 19px;
      left: 12px; }
  .toggle input {
    display: none; }
    .toggle input:checked + label {
      background-color: #60CA9B; }
      .toggle input:checked + label:before {
        transform: translateX(40px);
        box-shadow: -1px -1px 5px rgba(0, 0, 0, 0.1);
        animation: moveRight 0.3s ease-in-out; }

@keyframes moveRight {
  0% {
    width: 46px;
    transform: translateX(0); }
  45% {
    width: 52px; }
  100% {
    width: 46px;
    transform: translateX(40px); } }

@keyframes moveLeft {
  0% {
    width: 46px;
    transform: translateX(40px); }
  45% {
    width: 52px; }
  100% {
    width: 46px;
    transform: translateX(0); } }
