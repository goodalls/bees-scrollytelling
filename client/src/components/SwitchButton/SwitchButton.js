import React, { Component } from 'react';
import '../../assets/variables.scss';
import './SwitchButton.css';

class SwitchButton extends Component {
  constructor() {
    super();
    this.state = {
      checked: true
    }
  }

  handleChange = () => {
    this.setState({ checked: !this.state.checked })
    this.props.toggleQuestionsEnabled();
  }

  render() {
    return (
      <div className="toggle">
        <input checked={ this.state.checked } 
               onChange={this.handleChange}
               type="checkbox" 
               id="toggle" />
        <label htmlFor="toggle">
          <h4>Questions</h4>
          <div>
            <p>On</p>
            <p>Off</p>
          </div>
        </label>
      </div>
    )
  }
}  

export default SwitchButton; 