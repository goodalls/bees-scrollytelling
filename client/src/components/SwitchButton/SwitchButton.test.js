/* eslint-disable */
import React from 'react';
import SwitchButton from './SwitchButton';
import { shallow } from 'enzyme';

describe('SwitchButton', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<SwitchButton toggleQuestionsEnabled={jest.fn()} />);
  })

  it('matches the Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has default state', () => {
    const expected = {
      checked: true
    };

    expect(wrapper.instance().state).toEqual(expected);
  });

  describe('handleChange', () => {
    it('toggles checked value in state', () => {
      wrapper.instance().handleChange();

      expect(wrapper.instance().state.checked).toEqual(false);

      wrapper.instance().handleChange();

      expect(wrapper.instance().state.checked).toEqual(true);
    });

    it('calls toggleQuestionsEnabled', () => {
      expect(wrapper.instance().props.toggleQuestionsEnabled).not.toHaveBeenCalled();

      wrapper.instance().handleChange();

      expect(wrapper.instance().props.toggleQuestionsEnabled).toHaveBeenCalled();
    })
  })
});

