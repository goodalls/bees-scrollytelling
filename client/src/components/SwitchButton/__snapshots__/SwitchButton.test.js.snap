// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton matches the Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SwitchButton
    toggleQuestionsEnabled={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          checked={true}
          id="toggle"
          onChange={[Function]}
          type="checkbox"
/>,
        <label
          htmlFor="toggle"
>
          <h4>
                    Questions
          </h4>
          <div>
                    <p>
                              On
                    </p>
                    <p>
                              Off
                    </p>
          </div>
</label>,
      ],
      "className": "toggle",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": true,
          "id": "toggle",
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Questions
</h4>,
            <div>
              <p>
                            On
              </p>
              <p>
                            Off
              </p>
</div>,
          ],
          "htmlFor": "toggle",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Questions",
            },
            "ref": null,
            "rendered": "Questions",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  On
</p>,
                <p>
                  Off
</p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "On",
                },
                "ref": null,
                "rendered": "On",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Off",
                },
                "ref": null,
                "rendered": "Off",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            checked={true}
            id="toggle"
            onChange={[Function]}
            type="checkbox"
/>,
          <label
            htmlFor="toggle"
>
            <h4>
                        Questions
            </h4>
            <div>
                        <p>
                                    On
                        </p>
                        <p>
                                    Off
                        </p>
            </div>
</label>,
        ],
        "className": "toggle",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "id": "toggle",
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Questions
</h4>,
              <div>
                <p>
                                On
                </p>
                <p>
                                Off
                </p>
</div>,
            ],
            "htmlFor": "toggle",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Questions",
              },
              "ref": null,
              "rendered": "Questions",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p>
                    On
</p>,
                  <p>
                    Off
</p>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "On",
                  },
                  "ref": null,
                  "rendered": "On",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Off",
                  },
                  "ref": null,
                  "rendered": "Off",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
